### PR TITLE
Sondergaard/re organize code

### PR DESCRIPTION
**Pull request by Sondergaard**
## Description

- Some suggestions to the api surface
- Different way of creating a DatabricksSqlClient without mocking

### CODE REVIEW

File: source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs
Comment (HenrikSommer) [inline source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs:1]: This interface is growing rapidly - we should go through each method and evaluate what is needed. I have commented on some of them.
`IAsyncEnumerable<string[]> StreamAsync(string sqlStatement, List<SqlStatementParameter>? sqlStatementParameters);` can be removed when the new functionality is fully implemented.

File: source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs [hunk 1]
```diff
@@ -72,7 +73,57 @@ public interface IDatabricksSqlStatementClient
...
+    /// An asynchronous enumerable of <typeparamref name="T"/> object representing the result of the query.
+    /// </returns>
+    /// <remarks>
+    /// Use this method to execute SQL queries combined with Parameter Markers against Databricks to protect against SQL injection attacks.
+    /// The <paramref name="statement"/> should contain a collection of <see cref="QueryParameter"/>.
+    ///
+    /// Optionally, to simply execute a SQL query without parameters, the collection of <see cref="QueryParameter"/>
+    /// can be left empty. However, it is recommended to make use of parameters to protect against SQL injection attacks.
+    /// </remarks>
+    IAsyncEnumerable<T> ExecuteStatementAsync<T>(DatabricksStatement statement, Format format);
```
Comment (HenrikSommer) [inline source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs:102]: This method is apparently only in use by `public IAsyncEnumerable<T> ExecuteStatementAsync<T>(DatabricksStatement statement)` (current line 135), whish is not in use.

File: source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs [hunk 2]
```diff
@@ -81,5 +132,6 @@ public interface IDatabricksSqlStatementClient
     /// Optionally, to simply execute a SQL query without parameters, the collection of <see cref="QueryParameter"/>
     /// can be left empty. However, it is recommended to make use of parameters to protect against SQL injection attacks.
     /// </remarks>
-    IAsyncEnumerable<dynamic> ExecuteStatementAsync(Statement statement, Format format);
+    public IAsyncEnumerable<T> ExecuteStatementAsync<T>(DatabricksStatement statement)
```
Comment (HenrikSommer) [inline source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs:135]: Not in use?

File: source/Databricks/source/SqlStatementExecution/Client/DatabricksSqlStatmentClient.Statements.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+    private async IAsyncEnumerable<dynamic> DoExecuteStatementAsync(Abstractions.DatabricksStatement statement, Format format)
+    {
+        var strategy = format.GetStrategy(_options);
+        var request = strategy.GetStatementRequest(statement);
+        var sw = Stopwatch.StartNew();
+        var response = await request.WaitForSqlWarehouseResultAsync(_httpClient, StatementsEndpointPath);
+        // Metrics.RecordWarehouseDuration(sw.Elapsed);
```
Comment (HenrikSommer) [inline source/Databricks/source/SqlStatementExecution/Client/DatabricksSqlStatmentClient.Statements.cs:49]: I suggest we implement metrics in another PR (if at all).

**Review by HenrikSommer — APPROVED** [review]

File: source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs
Comment (Sondergaard) [reply source/Databricks/source/SqlStatementExecution/Abstractions/IDatabricksSqlStatementClient.cs:1]: @HenrikSommer - I agree. 

I would like to split this into two separate classes / interfaces. I will create a new PR for that change.

**Pull request merged**